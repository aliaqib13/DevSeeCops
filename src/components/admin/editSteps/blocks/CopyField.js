import React, { Component } from 'react';
import { Input } from 'antd';

const { TextArea } = Input;

export default class CopyField extends Component {
    constructor(props) {
        super(props);

        this.state = {
            type: 'CopyField',
            content: {
                text: '',
            },
        };
    }

    componentDidMount() {
        this.setState({
            ...this.props.block,
        });
    }

    inputChange = e => {
        this.setState({
            ...this.state,
            content: {
                text: e.target.value,
            },
        }, () => this.props.onChangeBlock(this.state, this.props.index));
    }

    render() {
        const { preview } = this.props;

        return (
            <div>
                <div className="block-item">
                    <TextArea
                        rows={4}
                        type="text"
                        name="text"
                        value={this.state.content.text}
                        onChange={this.inputChange}
                        disabled={preview}
                    />
                </div>
            </div>
        );
    }
}
