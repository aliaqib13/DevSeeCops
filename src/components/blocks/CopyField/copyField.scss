.copy-field {
  border: 1px solid #ececec;
  border-radius: 4px;
  background-color: #f8f9fa;
  padding: 15px 42px 15px 10px;
  position: relative;
  margin-bottom: 25px;
  .copy-button-container {
    position: absolute;
    right: 1px;
    top: 14px;
    .copy-button {
      border: none;
      background: none;
      outline: none;
      cursor: pointer;
      color: #1890ff;
      &:hover {
        color: #40a9ff;
      }
    }
  }
  .command {
    overflow: scroll;
    -ms-overflow-style: none;
    scrollbar-width: none;
    display: block;
        &-wrapper {
      white-space: nowrap;
    }
  }
  .command::-webkit-scrollbar {
    display: none;
  }
}

.copy-field.multiline {
  padding: 20px;
  .copy-button-container {
    right: 1px;
    top: 3px;
  }
  .command {
    &-wrapper {
      white-space: normal;
    }
  }
}