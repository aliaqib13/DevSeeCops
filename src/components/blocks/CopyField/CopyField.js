import React from 'react';
import Clipboard from 'react-clipboard.js';
import { Icon, Tooltip } from 'antd';
import './copyField.scss';

class CopyField extends React.Component {
    state = {
        tooltipTitle: 'Copy to clipboard',
    }

    onSuccess = e => {
        e.clearSelection();
        this.setState({
            tooltipTitle: 'Copied!',
        });
    }

    changeTooltipTitle = visible => {
        if (!visible) {
            this.setState({
                tooltipTitle: 'Copy to clipboard',
            });
        }
    }

    render() {
        const { content } = this.props;
        return (
            <div className='copy-field'>
                <Tooltip title={this.state.tooltipTitle} onVisibleChange={this.changeTooltipTitle}>
                    <span className="copy-button-container">
                        <Clipboard
                            onSuccess={this.onSuccess}
                            option-target={trigger => trigger.parentElement.nextElementSibling}
                            className="copy-button"
                        >
                            <Icon type="copy" />
                        </Clipboard>
                    </span>
                </Tooltip>

                <code className="command">
                    <span className="command-wrapper">
                        {content.text}
                    </span>
                </code>
            </div>
        );
    }
}

export default CopyField;
