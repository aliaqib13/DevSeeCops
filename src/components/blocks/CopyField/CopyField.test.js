import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CopyField from './CopyField';

describe('CopyField', () => {
    let component;
    const props = {
        content: {
            text: 'test',
        },
    };

    beforeEach(() => {
        component = shallow(<CopyField {...props} />);
    });

    it('Should render successfully', () => {
        const wrapper = component.find('.copy-field');
        expect(wrapper.exists()).toBeTruthy();
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should contain text which passed down by props', () => {
        const wrapper = component.find('.command-wrapper');
        expect(wrapper.text()).toBe(props.content.text);
    });

    it('Should has tooltip title "Copy to clipboard" on ClipboardButton', () => {
        const tooltip = component.find('Tooltip');
        expect(tooltip.props().title).toBe('Copy to clipboard');
    });
});
