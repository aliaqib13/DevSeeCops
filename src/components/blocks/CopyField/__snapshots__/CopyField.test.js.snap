// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyField Should render successfully 1`] = `
<div
  className="copy-field"
>
  <Tooltip
    arrowPointAtCenter={false}
    autoAdjustOverflow={true}
    mouseEnterDelay={0.1}
    mouseLeaveDelay={0.1}
    onVisibleChange={[Function]}
    placement="top"
    title="Copy to clipboard"
    transitionName="zoom-big-fast"
  >
    <span
      className="copy-button-container"
    >
      <ClipboardButton
        className="copy-button"
        isVisibleWhenUnsupported={true}
        onClick={[Function]}
        onSuccess={[Function]}
        option-target={[Function]}
      >
        <Icon
          type="copy"
        />
      </ClipboardButton>
    </span>
  </Tooltip>
  <code
    className="command"
  >
    <span
      className="command-wrapper"
    >
      test
    </span>
  </code>
</div>
`;
